module-check/src/main/clojure/clojure/core/typed/base_env.clj

```diff
@@ -740,6 +740,8 @@ clojure.string/trim [String -> String]
 clojure.string/trimr [String -> String]
 clojure.string/triml [String -> String]
 
+clojure.template/apply-template [ (t/HVec [t/Any *]) (t/HVec [t/Any *])(t/List t/Any ) -> (t/HVec [t/Any *])]
```

Comment:
Same comment about return value. It's not only vectors. It is the same as second argument.